pkg/config/config.go

```diff
@@ -148,47 +138,40 @@ func (operatorConfig *NSXOperatorConfig) validate() error {
 		return err
 	}
 	// TODO, verify if user&pwd, cert, jwt has any of them provided
-
-	return operatorConfig.validateVersion()
+	return nil
 }
 
+// it's not thread safe
 func (operatorConfig *NSXOperatorConfig) GetTokenProvider() auth.TokenProvider {
```

Comment:
Why do we still need to create/get token provider in config.go?

Reply:
it's used to NewConfig. since config only cares about TokenProvider, so create token provider in OperatorConfig.